python/lsst/rapid/analysis/imageSorter.py

```diff
@@ -0,0 +1,217 @@
+            print('Press S - skip all images with existing annotations, including blank annotations')
+            print('Press B - skip all images with annotations that are not blank')
+            print('Press D - just display existing data and exit')
+            print('Press Q to quit')
+            mode = input()
+            mode = mode[0].upper()
+
+            if mode == 'Q':
+                exit()
+            elif mode == 'D':
+                for dataId, value in info.items():
+                    print(f"{dataId[0]} - {dataId[1]}: {value}")
+                exit()
+            elif mode in 'AOSB':
+                pass
+            else:
+                print("Unrecognised response - try again")
+                self.sortImages()
+                return  # don't run twice in this case!
+
+        # need to write file first, even if empty, because _load and _save
+        # are inside the loop to ensure that annotations aren't lost even on
+        # full crash
+        print(TIPS)
+        self._save(info, self.outputFilename)
+
+        plt.figure(figsize=(10, 10))
+        for imNum, filename in enumerate(self.fileList):
+            info = self._load(self.outputFilename)
+
+            dataId = self._getDataIdFromFilename(filename)
+            if dataId in info and mode in ['S', 'B']:  # always skip if found for S and if not blank for B
+                if (mode == 'S') or (mode == 'B' and info[dataId] != ""):
+                    continue
+
+            with Image.open(filename) as pilImage:
+                pilImage = Image.open(filename)
+                width, height = pilImage.size
+                cropLR, cropUD = 100, 180
+                cropped = pilImage.crop((cropLR, cropUD, width-cropLR, height-cropUD))
+                plt.clf()
+                plt.imshow(cropped, interpolation="bicubic")
+                plt.show(block=False)
+                plt.draw()  # without this you get the same image each time
+                plt.tight_layout()
+                osascriptCall = '''/usr/bin/osascript -e 'tell app "Finder" to '''
+                osascriptCall += '''set frontmost of process "Terminal" to true' '''
+                system(osascriptCall)
+
+            oldAnswer = None  # just so we can display existing info with the dataId
+            if dataId in info:
+                oldAnswer = info[dataId]
+            inputStr = f"{dataId[0]} - {dataId[1]}: %s" % ("" if oldAnswer is None else oldAnswer)
+            answer = input(inputStr)
+            if 'exit' in answer:
+                break  # break don't exit so data is written!
+
+            self.addData(dataId, info, answer, mode, imNum)
+            self._save(info, self.outputFilename)
+
+        print(f'Info written to {self.outputFilename}')
+
+        return info
+
+
+if __name__ == '__main__':
+    fileList = ['/Users/merlin/rsync/animatorOutput/pngs/dayObs-2020-02-17-seqNum-232-calexp.png',
```

Comment:
Does this require editing this file to run on a different set of inputs?  Can those be command line options instead?  A CmdlineTask?